monai/apps/reconstruction/networks/nets/coil_sensitivity_model.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Union
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+
+from monai.apps.reconstruction.mri_utils import root_sum_of_squares
+from monai.apps.reconstruction.networks.nets.complex_unet import ComplexUnet
+from monai.networks.blocks.fft_utils_t import ifftn_centered_t
+
+
+class CoilSensitivityModel(nn.Module):
+    """
+    This class uses :py:class:`monai.apps.reconstruction.networks.nets.complex_unet` to learn
+    coil sensitivity maps for multi-coil MRI reconstruction. Learning is done on the center of
+    the under-sampled kspace (that region is fully sampled).
+
+    The data being a (complex) 2-channel tensor is a requirement for using this model.
+
+    Modified and adopted from: https://github.com/facebookresearch/fastMRI
+
+    Args:
+        spatial_dims: number of spatial dimensions.
+        features: six integers as numbers of features. denotes number of channels in each layer.
+        act: activation type and arguments. Defaults to LeakyReLU.
+        norm: feature normalization type and arguments. Defaults to instance norm.
+        bias: whether to have a bias term in convolution blocks. Defaults to True.
+        dropout: dropout ratio. Defaults to 0.0.
+        upsample: upsampling mode, available options are
+            ``"deconv"``, ``"pixelshuffle"``, ``"nontrainable"``.
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int = 2,
+        features: Sequence[int] = (32, 32, 64, 128, 256, 32),
+        act: Union[str, tuple] = ("LeakyReLU", {"negative_slope": 0.1, "inplace": True}),
+        norm: Union[str, tuple] = ("instance", {"affine": True}),
+        bias: bool = True,
+        dropout: Union[float, tuple] = 0.0,
+        upsample: str = "deconv",
+    ):
+        super().__init__()
+        self.unet = ComplexUnet(
+            spatial_dims=spatial_dims,
+            features=features,
+            act=act,
+            norm=norm,
+            bias=bias,
+            dropout=dropout,
+            upsample=upsample,
+        )
+
+    def chans_to_batch_dim(self, x: Tensor) -> Sequence:
```

Comment:
Would it be better to spell out `chans` to `channels`? Thanks!

Reply:
Have you checked monai Reshape? https://github.com/Project-MONAI/MONAI/blob/31c4209f1d84a95536b56c3e4d6f7b5a31fe9b6f/monai/networks/layers/simplelayers.py#L140

Reply:
Do you feel there will be other recon networks using this func? I was not sure. If you feel so, maybe put these funcs in `./nets/utils.py`.